applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -115,11 +115,13 @@ def GetDefaultSettings(cls):
             "compute_reactions": true,
             "block_builder" : true,
             "builder_and_solver_settings" : {
-                "consider_lagrange_multiplier_constraint_resolution" : "none",
                 "diagonal_values_for_dirichlet_dofs"                 : "use_max_diagonal",
-                "constraint_scale_factor"                            : "use_mean_diagonal",
-                "auxiliar_constraint_scale_factor"                   : "use_mean_diagonal",
-                "silent_warnings"                                    : false
+                "silent_warnings"                                    : false,
+                "advanced_settings"                                  : {
+                    "consider_lagrange_multiplier_constraint_resolution" : "none",
+                    "constraint_scale_factor"                            : "use_mean_diagonal",
+                    "auxiliar_constraint_scale_factor"                   : "use_mean_diagonal"
```

Comment:
But those are still weong for e.g. the elimB&S
I think you should just pass them on to the B&S for usage there

Reply:
The elimination B&S is not currently constructed with Parameters, but in case we add one we should add an empty "advanced_settings"

Reply:
I don’t think the manual manipulation is the way to go here

Reply:
Precisely this is the only way to avoid manual manipulation

Reply:
I don’t think so, the cleanest solution would be to just pass the parameters of the B&S to the B&S

Reply:
In that case the B&S with the same parameters will crash in the Parameters validation

Reply:
I propose the following:
~~~js
{
    ...
    "builder_and_solver_settings" : {
    // settings that are used by the structural-solver
    "advanced_builder_and_solver_settings" : { // maybe find a better name for this
        // this will be directly passed to the CTor of the B&S
    }
}
~~~

Reply:
Ok, but, what do I pass to the CTor if the validation may crash in some B&S?, do I call them directly "block_builder_and_solver_settings" and "lm_builder_and_solver_settings"?

Reply:
I would only Pass the params to the B&S that have this Ctor
And in the future to all of them

Reply:
My question is, ho do you make this parameters compatible with the constructor of other B&S, can you write an example?

Reply:
In here those settings would be empty, exactly the same how we do it with the linear solver
This is possible bcs we don’t do recursive validation

Reply:
Thats why I proposed the splitting into settings that go into the b&S and settings that are used by the solver

Reply:
But that's basically what I have done with the advance_settings

Reply:
Ok, I will update today later

Reply:
I took a look again and I want to make it more clear:
this is how it would look in the solver (names of the settings are just examples):
~~~json
"builder_and_solver_settings" : {
    "use_block_builder" : true,
    "use_lagrange_B&S" : false,
    "advanced_settings"                                  : {
        "diagonal_values_for_dirichlet_dofs"                 : "use_max_diagonal",
        "silent_warnings"                                    : false,
        "consider_lagrange_multiplier_constraint_resolution" : "none",
        "constraint_scale_factor"                            : "use_mean_diagonal",
        "auxiliar_constraint_scale_factor"                   : "use_mean_diagonal"
    }
}
~~~

Reply:
If I move the "use_block_builder" this will change current behavior, I would do that in a future PR